src/net_processing.cpp

```diff
@@ -1460,17 +1508,16 @@ void PeerManagerImpl::SendPings()
 
 void PeerManagerImpl::RelayTransaction(const uint256& txid, const uint256& wtxid)
 {
-    m_connman.ForEachNode([&txid, &wtxid](CNode* pnode) EXCLUSIVE_LOCKS_REQUIRED(::cs_main) {
-        AssertLockHeld(::cs_main);
-
-        CNodeState* state = State(pnode->GetId());
-        if (state == nullptr) return;
-        if (state->m_wtxid_relay) {
-            pnode->PushTxInventory(wtxid);
-        } else {
-            pnode->PushTxInventory(txid);
+    LOCK(m_peer_mutex);
+    for(auto& it : m_peer_map) {
+        Peer& peer = *it.second;
```

Comment:
That means node_id is unused, which causes warnings depending on your compiler flags.

Reply:
Which compiler flags produce a warning? I don't get any with either clang 12 or gcc 10:
```cpp
    std::map<int, int> m;
    for (const auto& [x, y] : m) { // no warning
        std::cout << x;
    } 
    for (const auto& [x, y] : m) { // warning: unused structured binding declaration [-Wunused-variable]
    } 
```

Reply:
Hmmm, if you search for "c++ structured binding unused variable", there do seem to be a lot of people claiming that they've encountered this warning. Like you, I've been unable to reproduce it, either locally or on godbold.org with various compiler versions.

In any case, I'm going to leave this as it is, since it's only one more line and no less clear to explicitly declare the `Peer` object. Thank you for pointing this out though!